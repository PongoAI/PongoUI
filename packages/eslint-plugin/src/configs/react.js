/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
