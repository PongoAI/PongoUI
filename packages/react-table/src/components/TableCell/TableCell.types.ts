import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type TableCellSlots = {
  /**
   * The root of the TableCell.
   */
  root: Slot<'th'>;
};

type TableCellCommons = {};

export type TableCellProps = ComponentProps<TableCellSlots> & TableCellCommons;

export type TableCellState = ComponentState<TableCellSlots> & TableCellCommons;
