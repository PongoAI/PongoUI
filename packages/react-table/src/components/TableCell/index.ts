export * from './TableCell';
export * from './useTableCellStyles';
export * from './TableCell.types';
export * from './useTableCell';
