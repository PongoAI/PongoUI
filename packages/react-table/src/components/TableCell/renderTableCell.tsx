import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { TableCellState, TableCellSlots } from './TableCell.types';

export const renderTableCell = (state: TableCellState) => {
  const { slots, slotProps } = getSlots<TableCellSlots>(state);

  return <slots.root {...slotProps.root} />;
};
