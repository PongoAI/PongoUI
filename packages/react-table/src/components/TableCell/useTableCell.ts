import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { TableCellProps, TableCellState } from './TableCell.types';

export const useTableCell = (props: TableCellProps, ref: React.Ref<HTMLElement>): TableCellState => {
  const state: TableCellState = {
    components: {
      root: 'th',
    },
    root: getNativeElementProps('th', {
      ref,
      ...props,
    }),
  };

  return state;
};
