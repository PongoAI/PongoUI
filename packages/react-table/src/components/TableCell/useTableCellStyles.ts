import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { TableCellState } from './TableCell.types';
import { tokens } from '@cebus/react-theme';

const useRootStyles = makeStyles({
  root: {
    display: 'table-cell',
    fontFamily: tokens.fontFamilyBase,
    verticalAlign: 'inherit',
    color: tokens.textColor,
    fontWeight: 'inherit',
    textAlign: 'left',
    ...shorthands.borderBottom('1px', 'solid', tokens.inheritBackground),
    ...shorthands.padding(tokens.focusedLayout),
  },
});

export const useTableCellStyles = (state: TableCellState) => {
  const rootStyles = useRootStyles();

  state.root.className = mergeClasses(rootStyles.root, state.root.className);

  return state;
};
