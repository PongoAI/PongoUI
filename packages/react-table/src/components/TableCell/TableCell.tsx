import * as React from 'react';
import { useTableCell } from './useTableCell';
import { useTableCellStyles } from './useTableCellStyles';
import { renderTableCell } from './renderTableCell';
import type { TableCellProps } from './TableCell.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The TableCell component is used to display a cell of data.
 */
export const TableCell: ForwardRefComponent<TableCellProps> = React.forwardRef((props, ref) => {
  const state = useTableCell(props, ref);

  useTableCellStyles(state);

  return renderTableCell(state);
});
TableCell.displayName = 'TableCell';
