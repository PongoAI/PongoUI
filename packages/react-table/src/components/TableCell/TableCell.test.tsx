import * as React from 'react';
import { TableCell } from './TableCell';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('TableCell', () => {
  isConformant({
    Component: TableCell,
    displayName: 'TableCell',
  });

  it('renders a basic Table', () => {
    const renderedComponent = render(<TableCell />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
