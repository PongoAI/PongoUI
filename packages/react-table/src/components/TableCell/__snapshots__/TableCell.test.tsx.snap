// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell renders a basic Table 1`] = `
<body>
  <div>
    <th
      class="___1tslpfj f15pt5es fk6fouc fvc3tfl fc467gg ff5ikls f1o700av f1f09k3d fg706s2 f18idrzn fk7qxxv f10sqwl7 f1eg49sl fne32i3"
    />
  </div>
</body>
`;
